src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -666,25 +666,41 @@ Status CoreWorkerDirectTaskSubmitter::CancelTask(TaskSpecification task_spec,
       [this, task_spec, scheduling_key, force_kill, recursive](
           const Status &status, const rpc::CancelTaskReply &reply) {
         absl::MutexLock lock(&mu_);
+        RAY_LOG(DEBUG) << "CancelTask RPC response received for " << task_spec.TaskId()
+                       << " with status " << status.ToString();
         cancelled_tasks_.erase(task_spec.TaskId());
 
-        if (status.ok() && !reply.attempt_succeeded()) {
-          if (cancel_retry_timer_.has_value()) {
-            if (cancel_retry_timer_->expiry().time_since_epoch() <=
-                std::chrono::high_resolution_clock::now().time_since_epoch()) {
-              cancel_retry_timer_->expires_after(boost::asio::chrono::milliseconds(
-                  RayConfig::instance().cancellation_retry_ms()));
+        // Retry is not attempted if !status.ok() because force-kill may kill the worker
```

Comment:
just to clarify. this doesn't change the logic but just move the code around right?

Reply:
No logic change. Just more logs for error case.